plugin/src/main/java/com/iridium/iridiumskyblock/listeners/PlayerPortalListener.java

```diff
@@ -9,24 +10,30 @@
 import org.bukkit.event.player.PlayerTeleportEvent;
 
 import java.util.Objects;
+import java.util.Optional;
 
 public class PlayerPortalListener implements Listener {
 
     @EventHandler
     public void onPlayerPortal(PlayerPortalEvent event) {
         IslandManager islandManager = IridiumSkyblock.getInstance().getIslandManager();
-        IridiumSkyblock.getInstance().getIslandManager().getIslandViaLocation(event.getFrom()).ifPresent(island -> {
-            event.setSearchRadius(island.getSize() / 2);
-            if (event.getCause().equals(PlayerTeleportEvent.TeleportCause.NETHER_PORTAL)) {
-                World world = Objects.equals(event.getFrom().getWorld(), islandManager.getNetherWorld()) ? islandManager.getWorld() : islandManager.getNetherWorld();
-                event.setTo(island.getCenter(world));
-            }
-            if (event.getCause().equals(PlayerTeleportEvent.TeleportCause.END_PORTAL)) {
-                event.setCancelled(true);
+        if ((event.getCause().equals(PlayerTeleportEvent.TeleportCause.NETHER_PORTAL) && IridiumSkyblock.getInstance().getConfiguration().netherIslands)
```

Comment:
That's a long if statement... Maybe declare two new variables?

Reply:
instead id just remove this if statement and put the if statement below the other if statement. idk how to describe it, ill just do it myself

Reply:
Like this: https://github.com/Iridium-Development/IridiumSkyblock/pull/99